File: js/anthropic/imageInput.ts
import Anthropic from "@anthropic-ai/sdk";
import "dotenv/config";
import { traceable } from "langsmith/traceable";
import { image } from "../resources/base64BufoApproval";

const client = new Anthropic();

const imageInputFromUrl = traceable(async function imageInputFromUrl(
  messages: Anthropic.Messages.MessageParam[]
) {
  console.log(
    "Analyzing image from URL (fetching and converting to base64)...\n"
  );

  const message = await client.messages.create({
    model: "claude-sonnet-4-5",
    max_tokens: 1024,
    messages,
  });

  console.log("Response:", message.content);
  return message;
});

const imageInputFromBase64 = traceable(async function imageInputFromBase64(
  messages: Anthropic.Messages.MessageParam[]
) {
  console.log("Analyzing image from base64...\n");

  const message = await client.messages.create({
    model: "claude-sonnet-4-5",
    max_tokens: 1024,
    messages,
  });

  console.log("Response:", message.content);
  return message;
});

const mainExample = traceable(async function mainExample() {
  const imageUrl =
    "https://upload.wikimedia.org/wikipedia/commons/thumb/d/dd/Gfp-wisconsin-madison-the-nature-boardwalk.jpg/2560px-Gfp-wisconsin-madison-the-nature-boardwalk.jpg";

  await imageInputFromUrl([
    {
      role: "user",
      content: [
        {
          type: "text",
          text: "Describe this image in detail:",
        },
        {
          type: "image",
          source: {
            // @ts-ignore
            type: "url",
            url: imageUrl,
          },
        },
      ],
    },
  ]);

  console.log("\n" + "=".repeat(50) + "\n");

  const base64Data = image.replace(/^data:image\/\w+;base64,/, "");

  await imageInputFromBase64([
    {
      role: "user",
      content: [
        {
          type: "text",
          text: "What do you see in this image? Describe it in detail.",
        },
        {
          type: "image",
          source: {
            type: "base64",
            media_type: "image/png",
            data: base64Data,
          },
        },
      ],
    },
  ]);
});

export async function main() {
  return mainExample();
}

export { imageInputFromBase64, imageInputFromUrl };

File: python/wrap_anthropic/image_input.py
import os
import sys

sys.path.insert(0, os.path.dirname(os.path.dirname(__file__)))

import anthropic
from helpers import fetch_image_base64
from langsmith import traceable
from langsmith.wrappers import wrap_anthropic

client = wrap_anthropic(anthropic.Anthropic())


@traceable(name="Image Input - Base64")
def image_input_base64():
    """Image input using base64 encoding."""
    image_url = "https://upload.wikimedia.org/wikipedia/commons/a/a7/Camponotus_flavomarginatus_ant.jpg"
    image_data = fetch_image_base64(image_url)

    response = client.messages.create(
        model="claude-sonnet-4-5",
        max_tokens=1024,
        messages=[
            {
                "role": "user",
                "content": [
                    {
                        "type": "image",
                        "source": {
                            "type": "base64",
                            "media_type": "image/jpeg",
                            "data": image_data,
                        },
                    },
                    {"type": "text", "text": "What is in this image?"},
                ],
            }
        ],
    )

    print(f"Image base64 response: {response.content[0].text[:100]}...")
    return response


@traceable(name="Image Input - URL")
def image_input_url():
    """Image input using URL."""
    image_url = "https://upload.wikimedia.org/wikipedia/commons/a/a7/Camponotus_flavomarginatus_ant.jpg"

    response = client.messages.create(
        model="claude-sonnet-4-5",
        max_tokens=1024,
        messages=[
            {
                "role": "user",
                "content": [
                    {"type": "image", "source": {"type": "url", "url": image_url}},
                    {"type": "text", "text": "What is in this image?"},
                ],
            }
        ],
    )

    print(f"Image URL response: {response.content[0].text[:100]}...")
    return response


def main():
    print("Running wrap_anthropic image input examples...")
    image_input_base64()
    image_input_url()
    return {"image_base64": "complete", "image_url": "complete"}
